streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -235,12 +242,35 @@ public String toString() {
     }
 
     /**
+     * Cleans up local state store directory ({@code state.dir}), by deleting all data with regard to the application-id.
+     * <p>
+     * May only be called either before instance is started or after instance is closed.
+     *
+     * @throws IllegalStateException if instance is currently running
+     */
+    public void cleanUp() {
```

Comment:
Should we also validate that `cleanUp()` will never remove the state directories of other applications that run in the same local environment?  We already had at least one case in the past where running multiple app instances caused problems w.r.t. state dir, so I think it makes sense to ensure that multiple app instances or (here) apps can co-exist peacefully in the same local environment.